test/integration/testStreamRows.js

```diff
@@ -462,3 +464,91 @@ describe('Test Stream Rows HighWaterMark', function ()
     });
   });
 });
+
+describe('Test stream row modes', function () {
+  this.timeout(18000);
+  let connection;
+  const sql = 'select * ' +
+    'from (select \'a\' as key, 1 as foo, 3 as name) as table1 ' +
+    'join (select \'a\' as key, 2 as foo, 3 as name2) as table2 ' +
+    'on table1.key = table2.key'
+  const expectedArray = ['a', 1, 3, 'a', 2, 3]
+  const expectedObject = {KEY_0: 'a', FOO_1: 1, NAME: 3, KEY_3: 'a', FOO_4: 2, NAME2: 3}
+
+  const createItCallback = function (testCase) {
+    return function (done) {
+      this.timeout(180000);
+      connection.execute({
+        sqlText: sql,
+        streamResult: true,
+        complete: function (err, stmt) {
+          testUtil.checkError(err);
+          var stream = stmt.streamRows({rowMode: testCase.rowMode});
+          stream.on('data', function (row) {
+            const mode = testCase.rowMode || testCase.connectionRowMode
+            if (mode === 'array') {
+              assert.ok(util.isArray(row))
+              assert.deepEqual(row, testCase.expected)
+            } else {
+              assert.ok(util.isObject(row))
+              assert.deepStrictEqual(row, testCase.expected)
+            }
+          }).on('end', function () {
+            done();
+          }).on('error', function (err) {
+            testUtil.checkError(err);
+          });
+        }
+      })
+    }
+  };
+
+  describe('Row mode - connection parameter', function () {
+    before(function (done) {
+      connection = testUtil.createConnection({rowMode: `array`});
+      testUtil.connect(connection, done);
+    });
+    after(function (done) {
+      testUtil.destroyConnection(connection, done);
+    });
+
+    it('test return result row as array - connection parameter',
+      createItCallback({
+        name: 'test return result as object with sufiix for duplicated columns',
+        connectionRowMode: 'array',
+        expected: expectedArray
+      }));
+  })
+
+  describe('Row mode - stream parameter', function () {
+    before(function (done) {
+      connection = testUtil.createConnection();
+      testUtil.connect(connection, done);
+    });
+
+    after(function (done) {
+      testUtil.destroyConnection(connection, done);
+    });
+
+    const testCases = [
```

Comment:
let's create tests where we set rowMode on two levels

Reply:
Tests added.